lib/server.js

```diff
@@ -44,85 +142,22 @@ module.exports = function (server, options) {
         server.on('wrappedConnection', listener);
     });
     server.removeAllListeners('connection');
-    server.on('connection', function (c) {
-        var length = null,
-            buffer = null;
-
-        function invalidHeader(message, data) {
-            var error = new Error(message);
-            error.type = data.type;
-            error.length = data.length;
-            if (server.listeners("headerError").length > 0) {
-                server.emit("headerError", error, c);
-            } else {
-                c.destroy();
-            }
-        }
-
-        function readable() {
-            if (!length) {
-                length = c.read(1);
-                if (length === null) {
-                    return;
-                }
-                length = length.readUInt8(0);
-                if (length === 0) {
-                    c.removeListener('readable', readable);
-                    return invalidHeader("Invalid Header Length", {length: length});
-                }
-            }
-            if (!buffer) {
-                buffer = c.read(length);
-                if (!buffer) {
-                    return;
-                }
-            }
-            c.removeListener('readable', readable);
-            var type = buffer.readUInt8(0);
-            switch (type) {
-            case 0:
-                if (length !== 5) {
-                    return invalidHeader("Invalid Header Length", {type: type, length: length});
-                }
-                (function () {
-                    var address = '',
-                        i;
-                    for (i = 0; i < 4; i = i + 1) {
-                        address = address + '.';
-                        address = address + buffer.readUInt8(i + 1).toString(10);
-                    }
-                    address = address.substring(1);
-                    address = ipaddr.parse(address).toString();
-                    manage(c, address);
-                }());
-                break;
-            case 1:
-                if (length !== 9) {
-                    return invalidHeader("Invalid Header Length", {type: type, length: length});
-                }
-                (function () {
-                    var address = '',
-                        i;
-                    for (i = 0; i < 8; i = i + 1) {
-                        address = address + ':';
-                        address = address + buffer.readUInt16BE(i * 2 + 1).toString(16);
-                    }
-                    address = address.substring(1);
-                    address = ipaddr.parse(address).toString();
-                    manage(c, address);
-                }());
-                break;
-            default:
-                return invalidHeader("Invalid Header Type", {type: type, length: length});
-            }
+    server.on('connection', function(c) {
+        c.on('readable', function(c) {
+            parser(c, manage);
             server.emit('wrappedConnection', c);
```

Comment:
This is not equivalent.
The `wrappedConnection` event was triggered at the end of the header parsing, which can require more than one `readable` event (the connection can be slow and not all the headers is returned).